File: language/java_study/javaLearn/src/main/java/com/iamhotcenplotz/www/chapter05/WhileLoop.java
package com.iamhotcenplotz.www.chapter05;

public class WhileLoop {
    public static void exercise01(){
        int i = 0;

        while (i <= 100){
            System.out.println(i);
            i++;
        }
    }

    public static void exercise02(){
        int begin = 40;
        int end = 200;
        while (begin <= end){
            if(begin % 3 == 0){
                System.out.println("begin: " + begin);
            }
            begin++;
        }
    }

    public static void main(String[] args) {
        exercise01();
        exercise02();
    }
}

File: language/java_study/javaLearn/src/main/scala/com/iamhotcenplotz/www/chapter05/WhileLoop_.scala
package com.iamhotcenplotz.www.chapter05

object  WhileLoop_ {
  def exercise01(): Unit = {
    var i: Int = 0

    while (i <= 100) {
      println(s"${i}")
      i += 1
    }
  }

  def doWhile(): Unit = {
    var i: Int = 0

    do {
      println(i)
      i += 1
    } while (i < 100)


  }

  def main(args: Array[String]): Unit = {
    exercise01()
    doWhile()

    val aList: List[String] = List("Hi", "How are you")
    println(
      aList.map(i => {
        s"${i} ${i}"
      }).map(j => s"${j} ~ ")
        .mkString
    )
  }

}
